common/src/main/java/com/google/auto/common/MoreElements.java

```diff
@@ -189,5 +195,24 @@ public static boolean isAnnotationPresent(Element element,
     return Optional.absent();
   }
 
+  /**
+   * Returns a {@link Predicate} that can be used to filter elements by {@link Modifier}.
```

Comment:
How about we name the method hasAllModifiers?

Maybe we should add a method hasExactModifiers?